src/monitor-api-requests.ts

```diff
@@ -130,11 +142,53 @@ export async function getExistingMonitors(
   const serverlessMonitorIdByMonitorId: { [key: string]: number } = {};
   for (const existingMonitor of existingMonitors) {
     for (const tag of existingMonitor.tags) {
-      if (tag.startsWith("serverless_monitor_id:")) {
+      if (tag.startsWith("serverless_monitor_id:") || tag.startsWith("serverless_id:")) {
         const serverlessMonitorId = tag.substring(tag.indexOf(":") + 1);
         serverlessMonitorIdByMonitorId[serverlessMonitorId] = existingMonitor.id;
       }
     }
   }
   return serverlessMonitorIdByMonitorId;
 }
+
+export async function getRecommendedMonitors(site: string, monitorsApiKey: string, monitorsAppKey: string) {
+  const recommendedMonitors: { [key: string]: ServerlessMonitor } = {};
+  const endpoint = `https://api.${site}/api/v2/monitor/recommended?count=50&start=0&search=tag%3A%22product%3Aserverless%22`;
+  const response: Response = await fetch(endpoint, {
+    method: "GET",
+    headers: {
+      "DD-API-KEY": monitorsApiKey,
+      "DD-APPLICATION-KEY": monitorsAppKey,
+      "Content-Type": "application/json",
+    },
+  });
+  if (response.status !== 200) {
+    throw new Error(`Can't fetch monitor params. Status code: ${response.status}. Message: ${response.statusText}`);
+  }
+
+  const json = await response.json();
+  const recommendedMonitorsData = json.data;
+  recommendedMonitorsData.forEach((recommendedMonitorParam: RecommendedMonitorParams) => {
+    const recommendedMonitor: ServerlessMonitor = {
+      name: recommendedMonitorParam.attributes.name,
+      threshold: recommendedMonitorParam.attributes.options.thresholds.critical,
+      message: recommendedMonitorParam.attributes.message,
+      type: recommendedMonitorParam.attributes.type,
+      query: (cloudFormationStackId: string, criticalThreshold: number) => {
+        let query = recommendedMonitorParam.attributes.query;
+        // replace $scope with cloudformation_stack_id
+        query = query.replace(/\$scope*/g, `aws_coudformation_stack-id:${cloudFormationStackId}`);
```

Comment:
I just noticed that the tag says aws_**coud**formation_stack-id! 😄 .. Glad I caught that or else these monitors would never work! Once you have some more changes for the comments, I'm going to try creating a monitor and see if it really alerts so that we can be sure they're working..

Reply:
Oops, good catch 😅